h2o-core/src/main/java/water/rapids/ast/prims/models/AstPermutationVarImp.java

```diff
@@ -0,0 +1,84 @@
+package water.rapids.ast.prims.models;
+
+import hex.Model;
+import water.Key;
+import water.Scope;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.rapids.Env;
+import water.rapids.PermutationVarImp;
+import water.rapids.Val;
+import water.rapids.ast.AstPrimitive;
+import water.rapids.ast.AstRoot;
+import water.rapids.vals.ValFrame;
+import water.util.TwoDimTable;
+
+import java.util.Locale;
+
+/**
+ * Ast class for passing the model, frame and metric to calculate Permutation Variable Importance
+ */
+public class AstPermutationVarImp extends AstPrimitive {
+
+    @Override
+    public int nargs()      { return 1 + 7; }
+
+    @Override
+    public String[] args()  { return new String[]{"model", "frame", "metric", "n_samples", "n_repeats", "features", "seed"}; }
+
+    @Override
+    public String str()     { return "PermutationVarImp"; }
+
+    @Override
+    public ValFrame apply(Env env, Env.StackHelp stk, AstRoot asts[]) {
+        Model model = stk.track(asts[1].exec(env)).getModel();
+        Frame fr = stk.track(asts[2].exec(env)).getFrame();
+        String metric = stk.track(asts[3].exec(env)).getStr().toLowerCase();
+        long n_samples = (long) stk.track(asts[4].exec(env)).getNum();
```

Comment:
Do you validate the parameters somewhere? For example, if `n_samples`  is higher than 1 and lower than dataset size? Or if the `n_repeats` is not negative. Or if the input features were used for model training...

Reply:
Good point, thank you! I completely missed that.